### PR TITLE
Overhaul

### PR DESCRIPTION
**Pull request by nullableVoidPtr**
Makefile just like emerald, among other niceties.

### CODE REVIEW

File: charmap.txt [hunk 1]
```diff
@@ -0,0 +1,217 @@
+' ' = 20
```
Comment (PikalaxALT) [inline charmap.txt]: wait isn't this just ascii? why do we need a charmap for firered?

File: data/text/luminous_cave.inc [hunk 2]
```diff
@@ -0,0 +1,137 @@
+gUnknown_80DCA34:: @ 80DCA34
+	.string "#+...#WA voice emanates from somewhere...#P#+...Ye who seek awak"
```
Comment (PikalaxALT) [inline data/text/luminous_cave.inc]: just use `.ascii` here
and fix the line break

File: charmap.txt (hunk 1 above)
Comment (nullableVoidPtr) [reply charmap.txt]: Yeah, it's redundant. Oversight on my part, will fix.

Comment (nullableVoidPtr) [reply charmap.txt]: Unable to match extended ASCII with `gas` `.ascii`.

**Pull request merged**